src/complex/DataStructure/DataStructure.cpp

```diff
@@ -517,6 +517,16 @@ DataStructure::ConstIterator DataStructure::end() const
 
 bool DataStructure::insert(const std::shared_ptr<DataObject>& dataObject, const DataPath& dataPath)
 {
+  if(dataObject == nullptr)
+  {
+    return false;
+  }
+
+  if(getData(dataObject->getId()) != nullptr)
+  {
+    dataObject->setId(generateId());
+  }
```

Comment:
If the object is already in the `DataStructure`, shouldn't it already have an id?

Reply:
What about adding an existing `DataObject` to a new path so that it would have multiple parents? It would overwrite the ID here.